src/library_memfs.js

```diff
@@ -276,16 +276,18 @@ mergeInto(LibraryManager.library, {
         // If memory can grow, we don't want to hold on to references of
         // the memory Buffer, as they may get invalidated. That means
         // we need to do a copy here.
+	if (buffer === HEAP8.buffer || buffer.buffer === HEAP8.buffer) {
```

Comment:
Shouldn't this always be a typed array view? Did you see a case where it can be an ArrayBuffer?

Reply:
I adapted this from the mmap code below
```
  (buffer === contents.buffer || buffer.buffer === contents.buffer) ) {
```
(=== order reversed for easier comparison)
I seems it will be a view in all cases, but I wasn't sure so I used the same kind of condition.
Should I test `buffer.buffer === HEAP8.buffer` only?

Reply:
I looked into this some more, and I do think it should always be a view. This seems to pass locally,

https://github.com/emscripten-core/emscripten/commit/623819b202a0257274cb9f32c8eb9d90c3ae4239

So yes, let's assume it's a view here. I an open a PR with that patch separately, or if you want you can bundle it here while updating this PR.

Reply:
OK, I prepended your commit and compared views only.